src/core/ee/vu.cpp

```diff
@@ -117,37 +125,47 @@ void VectorUnit::run(int cycles)
         cycles_to_run--;
     }
 
-    if (gif->path_active(1))
+    if (get_id() == 1)
     {
-        XGKICK_cycles++;
-        if (XGKICK_cycles >= 5)
+        if (gif->path_active(1))
```

Comment:
Because this code is getting long, the following code (after the PATH1 check passes) should be in its own function "handle_XGKICK".